corehq/apps/hqwebapp/utils/webpack.py

```diff
@@ -0,0 +1,21 @@
+import json
+from pathlib import Path
+
+import corehq
+
+WEBPACK_BUILD_DIR = Path(corehq.__file__).resolve().parent.parent / "webpack" / "_build"
+
+
+def get_webpack_manifest(filename=None):
```

Comment:
To address [this concern](https://github.com/dimagi/commcare-hq/pull/35104/files#r1750929823), what do you think of this solution?
```suggestion
from functools import wraps

from django.conf import settings


def cache_unless_debug(func):
    if settings.DEBUG:
        return func
    @wraps(func)
    def wrapper(*args, **kw):
        nonlocal cache
        if cache is None:
            cache = func(*args, **kw)
            assert cache is not None, f"{func} returned None"
        return cache
    cache = None
    return wrapper


@cache_unless_debug
def get_webpack_manifest(filename=None):
```


Then remove all `WEBPACK_MANIFEST` globals from corehq/apps/hqwebapp/templatetags/hq_shared_tags.py and instead call `get_webpack_manifest()` anywhere it's used.

Reply:
Follow-up because this has been nagging at me. If import time of this module is too early to access `settings.DEBUG` it should be possible to defer that check until later with a slightly more elaborate decorator. Let me know if an example would be helpful.

Reply:
updated. have been offline